src/components/organisms/UiMenu/useArrowNavigation.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  computed,
+  type ComponentInstance,
+  type Ref,
+} from 'vue';
+
+export interface ElementRef {
+  itemTemplateRefs?: ComponentInstance<any>;
+  isSelected: boolean;
+  isFocused: boolean;
+  tabindex: number;
+}
+
+export default function useArrowNavigation(elementsRefs: Ref<ElementRef[]>) {
+  const focusedElement = computed(() => elementsRefs.value.find((elementRef) => elementRef.isFocused));
+  const focusedElementIndex = computed(() => elementsRefs.value.findIndex((elementRef) => elementRef.isFocused));
+  const selectedElement = computed(() => (elementsRefs.value.find((elementRef) => elementRef.isSelected)));
+  const firstElement = computed(() => (elementsRefs.value.at(0)));
+  const initialElement = computed(() => (selectedElement.value || firstElement.value));
+  const lastElement = computed(() => (elementsRefs.value.at(-1)));
```

Comment:
I don't feel like this variable provides any value.

Reply:
It's used now to get to the last element of the list when the user uses "Page down" or "End".

Reply:
I understand. But I don't feel like using any object/array property should result in having a different variable.

Reply:
IMO it's not "any" array property. This property is last accessed element of menu. Then I want to provide it the same like `firstElement` or `initialElement` to share API for arrow navigation.

Reply:
I have no issue with `initialElement` - it does something non obvious
I have issue with doing
```
firstElement = array[0]
secondElement = array[1]
thirdElement = array[2]
```
Because it's self explanatory and unnecessary. 

I have similar issue with `focusedElementIndex`, it's unused and it's just a matter of doing `refs.indexOf(focusedElement.value)`.

Reply:
ok, but still I don't understand why I can create helper variables when I need it (this is the main problem in our discussion :))? 
1. Sure `focusedElementIndex` can be kept in `useArrowNavigation` without returning. But it is used in the `next` and `previous` elements. I need this value then I add var for it instead of using `refs.indexOf(focusedElement.value)`. Why I can't do that? 
2. `lastElement`, is defined in` useArrowNavigation`, arrow navigation allows you to jump to `first` and last `elements`. This var is in context (the context is arrowNavigation) then I still understand why I need to do something in the app instead of using this heleprs value.

Reply:
You're right about `focusedElementIndex` - I've been looking elsewhere and haven't noticed it's used right here. 🙈 
My bad!

About first/last:
```
const {
  firstElement,
} = useArrowNavigation(usedMenuTemplateRefs);

handleMenuItemFocus(firstElement);
```

Or

```
handleMenuItemFocus(usedMenuTemplateRefs.value.at(0));
```

I really don't know why I'd want to use a helper just for doing `.at(0)`. 🤷🏻 
I think it's very self-descriptive  that `.at(0)` means `First`.